releasing_guide.md

```diff
@@ -5,7 +5,10 @@ Cut a Release
 Create a PR to the `main` branch and go through the following steps:
 
 **Preparation**
-- Update and check changelog in `CHANGES.rst`: run `towncrier build` (to preview, run `towncrier build --draft`)
+- Prepare the release by running `python prepare_release.py vx.y` for a x.y release:
+  - Update the release notes in `CHANGES.rst`
+  - Update `setuptools_scm` fallback version
```

Comment:
Shouldn't this be done by the prepare_release.py function? We can remove about the release notes and the fallback version or perfereably just say that this are automatically changed in prepare_release.py.